tests/test_salesforce_sink.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Test Salesforce DataSource checkpoint functionality with CSV file streaming (Auto Loader style)
+"""
+import os
+import sys
+import shutil
+import time
+import tempfile
+import csv
+sys.path.append('.')
+
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col, lit
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, IntegerType
+
+def create_csv_file(file_path, data, headers):
+    """Create a CSV file with the given data"""
+    with open(file_path, 'w', newline='') as csvfile:
+        writer = csv.writer(csvfile)
+        writer.writerow(headers)
+        writer.writerows(data)
+
+def test_csv_checkpoint():
+    """Test checkpoint functionality using Salesforce streaming write"""
+    
+    # Check credentials
+    username = os.getenv('SALESFORCE_USERNAME')
+    password = os.getenv('SALESFORCE_PASSWORD')
+    security_token = os.getenv('SALESFORCE_SECURITY_TOKEN')
+    
```

Comment:
How to run these tests without the tokens? Maybe we can create an `examples` folder and add this as an example instead of a unit test `salesforce_sink_example.py`